compiler/hash-typecheck/src/storage/mod.rs

```diff
@@ -161,18 +166,23 @@ pub struct StorageRef<'tc> {
 
 /// A mutable reference to the storage, which includes both local and global
 /// storage, as well as core definitions.
-#[derive(Debug)]
-pub struct StorageRefMut<'tc> {
-    pub local_storage: &'tc mut LocalStorage,
-    pub global_storage: &'tc mut GlobalStorage,
-    pub source_map: &'tc SourceMap,
-}
+// #[derive(Debug)]
+// pub struct StorageRefMut<'tc> {
+//     pub local_storage: &'tc mut LocalStorage,
+//     pub global_storage: &'tc mut GlobalStorage,
+//     pub source_map: &'tc SourceMap,
+// }
 
 /// Trait that provides convenient accessor methods to various parts of the
 /// storage given a path to a [StorageRef] object.
 pub trait AccessToStorage {
     fn storages(&self) -> StorageRef;
 
+    /// Create a [TcDiagnosticsWrapper]
+    fn diagnostics(&self) -> TcDiagnosticsWrapper<Self> {
+        TcDiagnosticsWrapper(self)
+    }
```

Comment:
This should go into ops, it can also follow the pattern of taking StorageRef rather than some T: AccessToStorage.